mne/io/egi/egimff.py

```diff
@@ -672,6 +687,15 @@ def __init__(
         if len(annot["onset"]):
             self.set_annotations(Annotations(**annot, orig_time=None))
 
+        # create events from annotations
+        if events_as_annotations:
+            ev = find_events(self)
+            event_dict = {v: k for k, v in self.event_id.items()}
+            annot["onset"].extend(ev[:, 0] / self.info["sfreq"])
+            annot["duration"].extend(np.zeros(ev.shape[0]))
+            annot["description"].extend([event_dict[e] for e in ev[:, 2]])
+            self.set_annotations(Annotations(**annot))
```

Comment:
I would go a step further and when `events_as_annotations=True` is used, *do not* create the synthetic event channel and `raw.event_id`

Reply:
Sure, since those synthetic channels are already created by the time it reaches my code, I can delete them if `events_as_annotations` is true

Reply:
Rather than create then delete, would be better to avoid creation in the first place